### ISSUE TITLE
FrontdoorId doesn't return from AzureNative.Cdn.Profile

### ISSUE THREAD

**Issue by darena-patrick** [pulumi/pulumi-azure-native#1984]
### What happened?

FrontdoorId seems to not be properly mapped to its value. It returns null instead of the value.

### Steps to reproduce

I have tried both ways of accessing the FrontdoorId from the CDN Profile and they both return null:

```C#
var fdProfile = AzureNative.Cdn.GetProfile.Invoke(new AzureNative.Cdn.GetProfileInvokeArgs
{
    ProfileName = fd.Name,
    ResourceGroupName = rg.Name
});

var frontDoorId = fdProfile.Apply(v=>
{
    var frontDoorId = v.FrontdoorId;
    return Output.Create("frontDoorId");
});
```
```C#
fd.FrontdoorId.Apply(v => v);
```

### Expected Behavior

The FrontdoorId exists on the resource ... so it should return through Pulumi.

Here is the redacted resource:

```
{
    "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/xxxxxxxxxxxxxxx/providers/Microsoft.Cdn/profiles/xxxxxxxxxxxxxxx",
    "type": "Microsoft.Cdn/profiles",
    "name": "xxxxxxxxxxxxxxx",
    "location": "Global",
    "kind": "frontdoor",
    "sku": {
        "name": "Standard_AzureFrontDoor"
    },
    "properties": {
        "frontDoorId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
        "resourceState": "Active",
        "provisioningState": "Succeeded"
    }
}
```

### Actual Behavior

The value does not return, null value is returned instead.

### Additional context

I need this FrontdoorId so I can set the `x-azure-fdid` header on the IP restriction site configuration for a web app.
https://www.pulumi.com/registry/packages/azure-native/api-docs/web/webapp/#ipsecurityrestriction

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
Confirmed this also affects the CDN resource as well as the GetProfile invoke:

```typescript
export const profile = new azCdn.Profile("cdn-profile", {
  resourceGroupName: resourceGroup.name,
  sku: { name: "Standard_AzureFrontDoor" },
  location: "global",
});
```

Pulumi output:
```js
    profile   : {
        id               : "/subscriptions/*********/resourcegroups/resource-group*****/providers/Microsoft.Cdn/profiles/cdn-profile******"
        location         : "Global"
        name             : "cdn-profile*******"
        provisioningState: "Succeeded"
        resourceGroupName: "resource-group******"
        resourceState    : "Active"
        sku              : {
            name: "Standard_AzureFrontDoor"
        }
        type             : "Microsoft.Cdn/profiles"
        urn              : "urn:pulumi:dev::******::azure-native:cdn:Profile::cdn-profile"
    }
```

The output displayed is identical to the resource contained in the stack, so this appears to somewhere between the Azure API and the returning of fields from the provider (not the SDK).

**Comment by danielrbradley** [comment]
On further investigation this is simply resolve by explicitly using the latest version of the CDN API - `v20210601`. Change your code to use `AzureNative.Cdn.V20210601.GetProfile.Invoke(..)`.

This would indicate to me that's it's likely an upstream issue, specific to the `V20200901` (which is our current default).

**Comment by darena-patrick** [comment]
That did the trick. I noted between the 2 API versions that the property is `FrontDoorId` vs `FrontdoorId` ...

**Comment by velocitysystems** [comment]
Same issue here - `frontdoorId` returns null.

```
// Create Front Door CDN profile
const frontDoor = new azure.cdn.Profile(`cdnp-global-${stack}`, {
    resourceGroupName: resourceGroup.name,
    location: "global",
    sku: {
        name: azure.cdn.SkuName.Standard_AzureFrontDoor
    }
});
export const frontDoorId = frontDoor.frontdoorId;
```

**Comment by flo-sch** [comment]
Also experiencing the same issue (using TypeScript), I tried `v20210601`, `v20220501preview` and `v20221101preview`.
The `frontDoorId` output is defined, but its value is always undefined, preventing to set an access restriction on a WebApp instance:

```ts
const frontdoor = new azure.cdn.v20210601.Profile('frontdoor', {
  location: 'Global',
  resourceGroupName: resourceGroup.name,
  sku: {
    name: azure.cdn.v20210601.SkuName.Standard_AzureFrontDoor,
  }
});

const functionApp = new azure.web.WebApp('function-app', {
  // [...]
  siteConfig: {
    // [...]
    ipSecurityRestrictions: [
      {
        action: 'Deny',
        description: 'Deny all requests (default)',
        ipAddress: 'Any',
        name: 'Deny All',
        tag: azure.web.IpFilterTag.Default,
      },
      {
        action: 'Allow',
        description: 'Allow all requests through FrontDoor',
        headers: {
          'x-azure-fdid': [frontdoor.frontDoorId],
        },
        ipAddress: 'AzureFrontDoor.Backend',
        priority: 100,
        name: 'Allow through FrontDoor',
        tag: azure.web.IpFilterTag.ServiceTag,
      },
    ],
  },
});
```
  
> error: Code="BadRequest" Message="Headers is invalid.  The 'x-azure-fdid' value is empty!"

I would be curious to know how to call the suggested `AzureNative.Cdn.V20210601.GetProfile.Invoke(..)`, is such workaround available in TypeScript? 🙏

**Comment by flo-sch** [comment]
I managed to work around it using apply 🤷 

```ts
const functionApp = new azure.web.WebApp('function-app', {
  // [...]
  siteConfig: {
    // [...]
    ipSecurityRestrictions: [
      {
        action: 'Deny',
        description: 'Deny all requests (default)',
        ipAddress: 'Any',
        name: 'Deny All',
        tag: azure.web.IpFilterTag.Default,
      },
      {
        action: 'Allow',
        description: 'Allow all requests through FrontDoor',
        headers: frontdoor.frontDoorId.apply((id) => ({
          'x-azure-fdid': [id],
        })),
        ipAddress: 'AzureFrontDoor.Backend',
        priority: 100,
        name: 'Allow through FrontDoor',
        tag: azure.web.IpFilterTag.ServiceTag,
      },
    ],
  },
});
```

**Comment by mikhailshilkov** [comment]
This appears to be fixed in 2.0 because we use a modern API version 2023-05-01 by default. Verified with

```ts
import * as resources from "@pulumi/azure-native/resources";
import * as cdn from "@pulumi/azure-native/cdn";

const resourceGroup = new resources.ResourceGroup("resourceGroup");

const frontDoor = new cdn.Profile(`cdnp-global`, {
    resourceGroupName: resourceGroup.name,
    location: "global",
    sku: {
        name: cdn.SkuName.Standard_AzureFrontDoor
    }
});
export const frontDoorId = frontDoor.frontDoorId;
```

```
Outputs:
  + frontDoorId: "abc2146d-2a6a-43b2-96f0-bfe3f21cc23a"
```
